firmware/object.h

```diff
@@ -0,0 +1,50 @@
+// A generic object to use when linking different object classes
+
+#ifndef GENERICOBJECT
+#define GENERICOBJECT
+
+#include <ArduinoJson.h>
+#include <StreamUtils.h>
+#include <SoftwareSerial.h>
+#include <string.h>
+#include "interpreter.h"
+#include "json.h"
+#include "sensor.h"
+
+// A generic object that is inherited by specific object types to allow polymorphic
+// indexing of connected objects.
+// Functions in Generic Object are not defined so the object must be inherited before use
+class GenericObject{
+
+    protected:
+
+    // Structure that defines the key:value behavior which is compatible with JSON
+    typedef struct Attribute {
+        String name;
+        String value;
+
+    }Attribute;
+
+    // A container for multiple attributes that can be iterated through
+    typedef struct Attributes {
+        // Gives each object attributes (change size to fit the maximum needed objects)
+        const static int number = 10;
+        Attribute* attrs[number];
+    } Attributes;
+
+    public:
+
+    // Defines the function of the object while in operation
+    // Actiions in run will be run each time the run section of main executes
+    virtual int run();
```

Comment:
Question: do you want this function (and the next 2) to be virtual or pure-virtual functions? IOW, should this be:

virtual int run() = 0;

If you just want them to be virtual, then there should be an accompanying object.cpp here, I think

Reply:
Yes, it should. I was not aware of the difference but it should be pure virtual.